### PR TITLE
[Windows] Add checksum verification for AWS SAM CLI

### PR DESCRIPTION
**Pull request by erik-bershel**
# Description
Update installer script with checksum for AWS SAM CLI

#### Related issue:
https://github.com/github/c2c-actions-akvelon/issues/38

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/Installers/Install-AWS.ps1 [hunk 1]
```diff
@@ -13,6 +14,14 @@ Install-Binary -Url $sessionManagerUrl -Name $sessionManagerName -ArgumentList (
 $env:Path = $env:Path + ";$env:ProgramFiles\Amazon\SessionManagerPlugin\bin"
 
 # Install AWS SAM CLI
-Install-Binary -Url "https://github.com/awslabs/aws-sam-cli/releases/latest/download/AWS_SAM_CLI_64_PY3.msi" -Name "AWS_SAM_CLI_64_PY3.msi"
+$packageName = "AWS_SAM_CLI_64_PY3.msi"
+$repoURL = "https://github.com/awslabs/aws-sam-cli/releases/latest"
+$packagePath = Start-DownloadWithRetry -Url "$repoURL/download/$packageName" -Name $packageName
```
Comment (vpolikarpov-akvelon) [inline images/win/scripts/Installers/Install-AWS.ps1:19]: `repoURL` variable is used only once to create URL for download. Why not defining download URL instead and avoid extra string expansion?

```suggestion
$packageUrl = "https://github.com/awslabs/aws-sam-cli/releases/latest/download/$packageName"
$packagePath = Start-DownloadWithRetry -Url $packageUrl -Name $packageName
```

File: images/win/scripts/Installers/Install-AWS.ps1 [hunk 2]
```diff
@@ -13,6 +14,14 @@ Install-Binary -Url $sessionManagerUrl -Name $sessionManagerName -ArgumentList (
...
 
-Invoke-PesterTests -TestFile "CLI.Tools" -TestName "AWS"
\ No newline at end of file
+#region Supply chain security - AWS SAM CLI
+$fileHash = (Get-FileHash -Path $packagePath -Algorithm SHA256).Hash
+$externalHash = Get-HashFromGitHubReleaseBody -RepoOwner "awslabs" -RepoName "aws-sam-cli" -FileName $packageName
+Use-ChecksumСomparison $fileHash $externalHash
```
Comment (vpolikarpov-akvelon) [inline images/win/scripts/Installers/Install-AWS.ps1:24]:
```suggestion
Use-ChecksumComparison $fileHash $externalHash
```

**Review by vpolikarpov-akvelon — APPROVED** [review]

**Review by Alexey-Ayupov — APPROVED** [review]

**Review by Alexey-Ayupov — APPROVED** [review]

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**